### PR TITLE
Movement fps

### PR DESCRIPTION
**Pull request by Hevhove**
Quick fix for:

1. Enum for directions
2. Added movement based on elapsed frame_time
3. Adding the ESC exit again

### CODE REVIEW

File: includes/cub.h [hunk 1]
```diff
@@ -104,6 +104,11 @@ typedef struct s_cub {
 	t_pair_i		*step;
 	int				hit;
 	int				side;
+	/* textures */
+	double			wall_x;
+	t_pair_i		*texel;
```
Comment (arieivs) [inline includes/cub.h:139]: what's texel sorry? 😅

File: srcs/color.c [hunk 2]
```diff
@@ -5,11 +5,6 @@ int	get_trgb(int t, int r, int g, int b)
 	return (t << 24 | r << 16 | g << 8 | b);
 }
```
Comment (arieivs) [inline srcs/color.c:7]: why did you remove get_t ?

File: srcs/hooks/hooks.c [hunk 3]
```diff
@@ -58,21 +60,25 @@ int	key_down(int keycode, t_cub *cub)
 int update_display(t_cub *cub)
 {
 	static int frames = 0;
+	double frame_time;
 
-	if (frames++ == 1000)
+	cub->old_time = cub->time;
+	cub->time = get_time_micros();
+	frame_time = (double)(cub->time - cub->old_time) / 1000; // in milliseconds
+	if (frames++ == 15)
```
Comment (arieivs) [inline srcs/hooks/hooks.c:71]: 15 is too fast for me, it will break

File: srcs/raycasting/textures.c [hunk 4]
```diff
@@ -0,0 +1,63 @@
+#include "cub.h"
+
+t_dir_code	get_wall_dir(t_cub *cub)
+{
+	if (cub->side == 1 && cub->map_pos->y < cub->pos->y)
```
Comment (arieivs) [inline srcs/raycasting/textures.c]: I think we should make an enum for the side as well, that's actually something I had thought about already, but it wasn't that important cos it was only being used in a specific place. But now that it's being used in a completely different place, I think it's really important. Just like this I have no idea if the side 1 is x or y. So we could just do an enum with SIDE_X and SIDE_Y to be 0 and 1

File: srcs/raycasting/textures.c [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+		return (NO);
+	if (cub->side == 1 && cub->map_pos->y > cub->pos->y)
+		return (SO);
+	else if (cub->side == 0 && cub->map_pos->x < cub->pos->x)
```
Comment (arieivs) [inline srcs/raycasting/textures.c]: why using else here?

File: srcs/raycasting/textures.c [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+		cub->texel_pos += cub->texel_step;
+		wall_dir = get_wall_dir(cub);
+		color = get_texture_color(cub->walls[wall_dir].tex, cub->texel->x, cub->texel->y);
+	}
+	else
+		color = cub->floor_color;
+	my_mlx_pixel_put(cub, x, i, color);
+}
+
+void	apply_textures(t_cub *cub, int x)
```
Comment (arieivs) [inline srcs/raycasting/textures.c:109]: I will want some in person explanations aha, I also really need to read the texture part on lode's tutorial

File: includes/cub.h (hunk 1 above)
Comment (Hevhove) [reply includes/cub.h:139]: A texel is a pixel on a texture. It's important to differentiate this as texels need to be translated correctly into pixel values (stretching them or skipping some of the texels).

File: srcs/color.c (hunk 2 above)
Comment (Hevhove) [reply srcs/color.c:7]: We're not using it at all because no objects are transparent. I doubt Minilibx can even handle transparency actually?
Also this way this file has max 5 functions!

File: srcs/hooks/hooks.c (hunk 3 above)
Comment (Hevhove) [reply srcs/hooks/hooks.c:71]: I can start doing some research on this because I am now at school and can test between both OS. But it seems there are some significant differences I hope I can fix..

Just to clarify as well: the frames variable is used so we don't do the calculation of the rays every mlx_loop_hook. It seems to me that as raycasting is an old technique, it runs extremely fast on modern computers and is also why our FPS is in the 10,000s. If we were implementing a raytracer instead, our FPS would probably be much lower as every individual pixel needs to be animated rather than every column of pixels.

File: srcs/raycasting/textures.c (hunk 5 above)
Comment (Hevhove) [reply srcs/raycasting/textures.c]: Ok cleaned it up.

File: srcs/raycasting/textures.c (hunk 6 above)
Comment (Hevhove) [reply srcs/raycasting/textures.c:109]: Yes I'll be doing my best to add comments today!

File: includes/cub.h (hunk 1 above)
Comment (arieivs) [reply includes/cub.h:139]: oh so that's really a thing? could you add that explanation (just the texel = pixel on a texture part) to the comment?

File: srcs/color.c (hunk 2 above)
Comment (arieivs) [reply srcs/color.c:7]: the mac version of minilibx can definetely handle transparency, which is why everything was transparent on your mac at the beginning :p maybe the linux version of minilibx can't, that I don't know. the limit of functions per file alone shouldn't be a strong enough argument, we can just create another file

File: srcs/hooks/hooks.c (hunk 3 above)
Comment (arieivs) [reply srcs/hooks/hooks.c:71]: Truth is, my computer was hyperventilating everytime I ran our cub programme, and it stoped doing that with this new implementation of yours. So for sure it is optimized. I just think we need to rethink that number over there. As I said, the best one for my computer is 100. But I would like to make it dependent on the frame_time, or from some other variable that may change with the device, yet my attempts failed.

File: srcs/raycasting/textures.c (hunk 6 above)
Comment (arieivs) [reply srcs/raycasting/textures.c:109]: thanks 🙏

File: srcs/color.c (hunk 2 above)
Comment (Hevhove) [reply srcs/color.c:7]: Fair enough on the limit of functions/minilibx, but why let a function in which is never called?

Comment (arieivs) [reply srcs/color.c:7]: It was being called 😂 It's as useful/useless (that's discussable I agree) as the other 3 functions (get_r/g/b)

File: srcs/raycasting/textures.c (hunk 4 above)
Comment (Hevhove) [reply srcs/raycasting/textures.c]: Done.

File: srcs/color.c (hunk 2 above)
Comment (Hevhove) [reply srcs/color.c:7]: I mean only in the print.. but ok added it back!

File: includes/cub.h (hunk 1 above)
Comment (Hevhove) [reply includes/cub.h:139]: Done.

File: includes/cub.h [hunk 7]
```diff
@@ -213,6 +234,7 @@ char				*set_fps_string(char *str1, t_cub *cub, int mode);
 int update_display(t_cub *cub);
 int	key_up(int keycode, t_cub *cub);
 int	key_down(int keycode, t_cub *cub);
+int	exit_game(int keycode, t_cub *cub);
```
Comment (arieivs) [inline includes/cub.h]: Did you really use this function in the end? I don't see it

Comment (Hevhove) [reply includes/cub.h]: Nope, it's a remnant from the old_days 👵🏻 , now that we have our friend graceful_exit 👑. Removing it
